bgpd/bgp_vty.c

```diff
@@ -1684,47 +1763,60 @@ static int bgp_update_delay_deconfig_vty(struct vty *vty)
 
 void bgp_config_write_update_delay(struct vty *vty, struct bgp *bgp)
 {
-	if (bgp->v_update_delay != BGP_UPDATE_DELAY_DEF) {
+	/* If configured globally, no need to display per-instance value */
+	if (bgp->v_update_delay != bm->v_update_delay) {
 		vty_out(vty, " update-delay %d", bgp->v_update_delay);
 		if (bgp->v_update_delay != bgp->v_establish_wait)
 			vty_out(vty, " %d", bgp->v_establish_wait);
 		vty_out(vty, "\n");
 	}
 }
 
+/* Global update-delay configuration */
+DEFPY (bgp_global_update_delay,
+       bgp_global_update_delay_cmd,
+       "bgp update-delay (0-3600)$delay [(1-3600)$wait]",
```

Comment:
If you're adding a new command (or new options) it might be a good idea to update the BGP YANG module accordingly to keep it in sync with the CLI.

Reply:
I'm a bit confused about how to do this.  The existing grouping called "global-config-timers" in yang/frr-bgp-common.yang contains both the rmap-delay-timer (which is defined globally in vtysh) and update-delay-time (which before this PR was defined per-vrf.)  The only change I'm making in this PR to allow a global config knob to set the update-delay value in all vrfs/instances.  I'll reach out to Vishal and Santosh who did the bgp yang model to try to understand the right thing to do.

Reply:
Chatted with Vishal and there currently isn't a "global bgp" table to hold these sort of items.  He is adding it and will include the update-delay command in the yang model.

Reply:
@dslicenc awesome. Thanks for looking at this.